scripts/bigquery_to_parquet.py

```diff
@@ -0,0 +1,90 @@
+
+
+import click
+import sys
+import pyarrow.parquet as pq
+import geopandas as gpd
+
+from encoder import AVAILABLE_COMPRESSIONS, Edges, PathType, geopandas_to_arrow
+from pathlib import Path
+from google.cloud import bigquery
+
+MODES = ["FILE", "FOLDER"]
+
+def read_gdf(input_query: str, primary_column: str):
+    client = bigquery.Client()
+    df = client.query(input_query).to_dataframe()
+    df[primary_column] = gpd.GeoSeries.from_wkt(df[primary_column])
+    return gpd.GeoDataFrame(df, geometry=primary_column, crs="EPSG:4326")
```

Comment:
This is a big, unnecessary bottleneck. Assuming that you're using recent versions of the python bigquery client library, bigquery downloads the data as Arrow. So in effect this makes two extra copies of the entire dataset: first to copy into Numpy/GEOS memory and again to copy back into Arrow memory. If the only purpose is to write to geoparquet, you can just parse the WKT to WKB, keep it in Arrow memory, and manually add metadata

Reply:
(This would require a little more custom code than the existing approach that reuses the existing `geopandas_to_arrow` function, but would be worth it if performance is important)